mockgen/parse.go

```diff
@@ -512,6 +526,38 @@ func iterInterfaces(file *ast.File) <-chan namedInterface {
 	return ch
 }
 
+type namedFunc struct {
+	name *ast.Ident
+	ft   *ast.FuncType
+}
+
+// Create an iterator over all function types in file.
+func iterFuncs(file *ast.File) <-chan namedFunc {
+	ch := make(chan namedFunc)
+	go func() {
```

Comment:
Can we return a slice instead of a channel? I don't think we need the extra complexity of a go routine and channels here.

Reply:
Though I agree returning a slice would also be acceptable here, this was done to be consistent with `iterInterfaces()` (placed right above `iterFuncs()`).

Would you want me to convert both of these to using a slice? I guess we should also rename these functions then, as they then don't iterate. `getInterfacesInFile()` and `getFuncsInFile()`?

Reply:
I think I would prefer we improve the codebase instead of continuing questionable patterns. WDYTA a refactor commit?

Reply:
Done!